frontend/src/components/Psychologist/Reimbursement/Billing.jsx

```diff
@@ -25,37 +25,20 @@ const Billing = () => {
 
   const { userStore: { user } } = useStore();
 
-  const [valuesByDate, setValuesByDate] = useState({ appointments: {}, patients: {} });
+  const [valuesByDate, setValuesByDate] = useState({ appointments: {}, firstAppointments: {}, patients: {} });
   const [fillInfo, setFillInfo] = useState(false);
   const [billingInfo, setBillingInfo] = useState(billingInfoService.get());
 
-  useEffect(() => {
-    agent.Appointment.get().then(response => {
-      const appointmentsByDate = {};
-      const patientsByDate = {};
-      response.forEach(appointment => {
-        const existingAppointments = appointmentsByDate[appointment.appointmentDate];
-        appointmentsByDate[appointment.appointmentDate] = existingAppointments ? existingAppointments + 1 : 1;
-        const existingPatients = patientsByDate[appointment.appointmentDate];
-        if (existingPatients) {
-          existingPatients.push(appointment.patientId);
-        } else {
-          patientsByDate[appointment.appointmentDate] = [appointment.patientId];
-        }
-      });
-      setValuesByDate({
-        appointments: appointmentsByDate,
-        patients: patientsByDate,
-      });
-    });
-  }, []);
+  useCommonEffect(setValuesByDate);
 
-  const filteredDate = Object.keys(valuesByDate.appointments).filter(date => {
+  const getFilteredDates = dates => Object.keys(dates).filter(date => {
```

Comment:
:green_circle: 
```suggestion
  const getFilteredDates = valuesByDate => Object.keys(valuesByDate).filter(date => {
```
Même remarque que plus haut